lib/active_merchant/billing/gateways/pay_arc.rb

```diff
@@ -0,0 +1,345 @@
+
+      def verify(creditcard, options = {})
+        token(creditcard, options)
+      end
+
+      #:nodoc:
+      def token(creditcard, options = {})
+        post = {}
+        post['authorize_card'] = 1
+        post['card_source'] = options[:card_source]
+        add_creditcard(post, creditcard, options)
+        add_address(post, creditcard, options)
+        post = options.update(post)
+        commit(STANDARD_ACTIONS[:token], post)
+      end
+
+      def supports_scrubbing? #:nodoc:
+        true
+      end
+
+      def scrub(transcript)
+        #:nodoc:
+        transcript.
+          gsub(%r((Authorization: Bearer )[^\s]+\s)i, '\1[FILTERED]\2').
+          gsub(%r((&?card_number=)[^&]*)i, '\1[FILTERED]').
+          gsub(%r((&?cvv=)[^&]*)i, '\1[BLANK]')
+      end
+
+      private
+
+      def charge(money, authorization, options = {})
+        post = {}
+        post['token_id'] = authorization
+        post['capture'] = options[:capture] || 1
+        add_money(post, money, options)
+        commit(STANDARD_ACTIONS[:capture], post)
+      end
+
+      def add_creditcard(post, creditcard, options)
+        post['card_number'] = creditcard.number
+        post['exp_month'] = creditcard.month
+        post['exp_year'] = creditcard.year
+        post['cvv'] = creditcard.verification_value unless creditcard.verification_value.nil?
+        post['card_holder_name'] = options[:card_holder_name] || "#{creditcard.first_name} #{creditcard.last_name}"
+      end
+
+      def add_address(post, creditcard, options)
+        post['address_line1'] = options[:address_line1]
+        post['address_line2'] = options[:address_line2]
+        post['city'] = options[:city]
+        post['state'] = options[:state]
+        post['zip'] = options[:zip]
+        post['country'] = options[:country]
+      end
+
+      def add_money(post, money, options)
+        post['amount'] = money
+        post['currency'] = (options[:currency] || currency(money))
+        post['statement_description'] = options[:statement_description]
+      end
+
+      def headers(api_key)
+        {
+          'Authorization' => 'Bearer ' + api_key.strip,
+          'Accept' => 'application/json',
+          'User-Agent' => "PayArc ActiveMerchantBindings/#{ActiveMerchant::VERSION}"
+        }
+      end
+
+      def parse(body)
+        JSON.parse(body)
+      rescue JSON::ParserError
+        body
+      end
+
+      def commit(action, parameters)
+        url = (test? ? test_url : live_url)
+        headers = headers(@options[:api_key])
+        end_point = "#{url}/#{action}"
+        begin
+          response = ssl_post(end_point, post_data(action, parameters), headers)
+          parsed_response = parse(response)
+
+          Response.new(
+            success_from(parsed_response, action),
+            message_from(parsed_response, action),
+            parsed_response,
+            test: test?,
+            authorization: parse_response_id(parsed_response)
```

Comment:
The `error_code_from` message is not implemented currently in the PayArc Active Merchant code. This causes the incorrect value to be passed back to Spreedly during error testing.

Please update the PayArc Active Merchant code with the `error_code_from` method implemented in the commit Response. You can find an example here: https://github.com/activemerchant/active_merchant/blob/2c2f70843f9043401398308f57f7c7986cf90d81/lib/active_merchant/billing/gateways/d_local.rb#L177

Reply:
Added this and pushed the code. Please check

Reply:
> @cdm-83 It looks like there are a good deal of merge conflicts here - please rebase with master, resolve the conflicts, and squash to one commit with your changes so I can see them easily. Thank you!

@jessiagee - 'Rebased again, and lots of conflicts in others file as well. Tried resolving those created a single commit. Let me if this commit looks fine. If not will create a new fork from the main repo and will merge the changes.

Senthil

Reply:
Hey @senthil-code thanks for doing this, but I think you're correct. 

My suggestion is to close this PR, get latest AM changes, then open a new PR with your PayArc updates so that we can easily see the changes and approve/merge them there.

Reply:
Hi @jessiagee 

Opened a new PR, please check this,

https://github.com/activemerchant/active_merchant/pull/4021